json_annotation/lib/src/json_serializable.dart

```diff
@@ -172,20 +172,11 @@ class JsonSerializable {
   ///
   /// If a field is annotated with `JsonKey` with a non-`null` value for
   /// `includeIfNull`, that value takes precedent.
-  final bool includeIfNull;
-
-  /// When `true` (the default), `null` fields are handled gracefully when
-  /// encoding to JSON and when decoding `null` and nonexistent values from
-  /// JSON.
-  ///
-  /// Setting to `false` eliminates `null` verification in the generated code,
-  /// which reduces the code size. Errors may be thrown at runtime if `null`
-  /// values are encountered, but the original class should also implement
-  /// `null` runtime validation if it's critical.
-  final bool nullable;
```

Comment:
I am a bit unsure that this should really be dropped at this time. Legacy projects may want to update their deps like analyzer, and thus require the latest json_serializable, but then no longer have a way of annotating their types as nullable since they are legacy still?

Up to you ultimately.

Reply:
Hrm. I left the entries in the constructor, but I removed the fields. So for the most common usage, folks aren't blocked.

We can always come back and publish a before-the-break version of json_serial if folks want it w/ latest analyzer

Reply:
There is definitely a lot of appeal to not trying to support weird mixed version stuff.

I also noticed after I left this that you already throw if running on a non-opted in library, so if it is a clean break that SGTM.

Reply:
ack